app/helpers/projects_helper.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+# View Helpers for Projects, Project Violations
+module ProjectsHelper
+  def violation_table_headers(project)
+    [
+      nil, # column for checkbox
+      "#{:CONSTRAINTS.l}:",
+      "#{:DATES.l}: #{project.date_range}",
+      "Lat: #{project.location.north} to #{project.location.south}",
+      "Lon: #{project.location.west} to #{project.location.east} ",
+      location_link(project.location.display_name, project.location,
+                    nil, false),
+      nil # column for observation.user
+    ]
+  end
+
+  def violation_table_rows(form:, project:, violations:)
+    violations.each_with_object([]) do |obs, rows|
+      rows << [
+        violation_checkbox(form: form, project: project, obs: obs),
+        link_to_object(obs, obs.text_name) + " (#{obs.id})",
+        styled_obs_when(project, obs),
+        styled_obs_lat(project, obs),
+        styled_obs_long(project, obs),
+        styled_obs_where(project, obs),
+        user_link(obs.user)
+      ]
+    end
+  end
+
+  def violations_help_text(project)
+    :form_violations_help.l if project.violations_removable_by_current_user?
+  end
+
+  def violations_submit_text(project)
+    if project.violations_removable_by_current_user?
+      :SUBMIT.l
```

Comment:
I think it would be better if this said "Remove Selected" since that's really the action it will take.  There's a comment at the top but that's pretty remote from the button.

Reply:
Also, if there are no violations it would be better if it said "Show Project".